### PR TITLE
feat(button) combine boolean fill/size/expand props to string

### PR DESCRIPTION
**Pull request by camwiegert**
#### Short description of what this resolves:
This combines the several sets of mutually-exclusive boolean props belonging to `ion-button` into a single string prop.

#### Changes proposed in this pull request:
- Combine `small, large, default` props under `size`. (e.g. `size="small"`)
- Combine `outline, clear, solid` props under `fill`. (e.g. `fill="outline"`)
- Combine `full, block` props under `expand`. (e.g. `expand="full"`)

**Ionic Version**: 4.x

**Fixes**: #13245

### CODE REVIEW

File: packages/core/src/components/button/button.tsx [hunk 1]
```diff
@@ -88,45 +77,21 @@ export class Button {
...
-  @Prop() clear: boolean = false;
-
-  /**
-   * @input {boolean} If true, activates a solid button style. Normally the default, useful for buttons in a toolbar.
-   * Type: style
+   * @input {string} Set to `"clear"` for a transparent button style or to `"outline"` for a transparent
+   * button style with a border. The default is `"solid"`.
    */
-  @Prop() solid: boolean = false;
+  @Prop() fill: ButtonFill = 'default';
```
Comment (brandyscarney) [inline packages/core/src/components/button/button.tsx]: Should the default here be `'solid'`?

File: packages/core/src/components/button/button.tsx [hunk 2]
```diff
@@ -252,55 +201,38 @@ function getClassList(buttonType: string, type: string, mode: string): string[]
...
-  let className =
-    buttonType +
-    ((style && style !== 'default') ?
-      '-' + style.toLowerCase() :
-      '');
+  if (buttonType !== 'bar-button' && fill === 'solid') {
+    fill = 'default';
+  }
+
+  if (fill && fill !== 'default') {
```
Comment (brandyscarney) [inline packages/core/src/components/button/button.tsx]: Same with this - are we supporting `'default'` as a fill, shouldn't it just be `'solid'`?

**Comment by manucorporat** [comment]
I think `fill` can be kind of confusing, since it is basically a color rule in CSS for svg.

File: packages/core/src/components/button/button.tsx [hunk 3]
```diff
@@ -27,45 +76,21 @@ export class Button {
...
-   * @input {boolean} If true, activates the large button size.
-   * Type: size
-   */
-  @Prop() large: boolean = false;
-
-  /**
-   * @input {boolean} If true, activates the small button size.
-   * Type: size
+   * @input {string} The button size.
+   * Possible values are: `"small"`, `"large"`, `"default"`.
```
Comment (brandyscarney) [inline packages/core/src/components/button/button.tsx]: is `"default"` an option? I don't see it below.

File: packages/core/src/components/button/button.tsx [hunk 4]
```diff
@@ -27,45 +76,21 @@ export class Button {
...
-   * @input {boolean} If true, activates a transparent button style without a border.
-   * Type: style
-   */
-  @Prop() clear: boolean = false;
-
-  /**
-   * @input {boolean} If true, activates a solid button style. Normally the default, useful for buttons in a toolbar.
-   * Type: style
+   * @input {string} Set to `"clear"` for a transparent button style or to `"outline"` for a transparent
+   * button style with a border. The default is `"solid"`.
```
Comment (brandyscarney) [inline packages/core/src/components/button/button.tsx]: Says the default is `"solid"` but it is set to `"default"` below. We should probably describe that the default is `"solid"` for normal buttons and `"clear"` for navbar buttons.

File: packages/core/src/components/button/button.tsx (hunk 3 above)
Comment (camwiegert) [reply packages/core/src/components/button/button.tsx]: Yeah, `"default"` shouldn't be listed there. Will remove.

**Pull request merged**